# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-core/src/command/encoder_command.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/wgpu-core/src/command/encoder_command.rs
--- a/third_party/rust/wgpu-core/src/command/encoder_command.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/wgpu-core/src/command/encoder_command.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,8 +1,11 @@
 use core::convert::Infallible;
 
-use alloc::{string::String, vec::Vec};
+use alloc::{string::String, sync::Arc, vec::Vec};
 
-use crate::id;
+use crate::{
+    id,
+    resource::{Buffer, QuerySet, Texture},
+};
 
 #[derive(Clone, Debug)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
@@ -68,3 +71,71 @@
         tlas: Vec<crate::ray_tracing::TraceTlasPackage>,
     },
 }
+
+#[derive(Clone, Debug)]
+pub enum ArcCommand {
+    CopyBufferToBuffer {
+        src: Arc<Buffer>,
+        src_offset: wgt::BufferAddress,
+        dst: Arc<Buffer>,
+        dst_offset: wgt::BufferAddress,
+        size: Option<wgt::BufferAddress>,
+    },
+    CopyBufferToTexture {
+        src: wgt::TexelCopyBufferInfo<Arc<Buffer>>,
+        dst: wgt::TexelCopyTextureInfo<Arc<Texture>>,
+        size: wgt::Extent3d,
+    },
+    CopyTextureToBuffer {
+        src: wgt::TexelCopyTextureInfo<Arc<Texture>>,
+        dst: wgt::TexelCopyBufferInfo<Arc<Buffer>>,
+        size: wgt::Extent3d,
+    },
+    CopyTextureToTexture {
+        src: wgt::TexelCopyTextureInfo<Arc<Texture>>,
+        dst: wgt::TexelCopyTextureInfo<Arc<Texture>>,
+        size: wgt::Extent3d,
+    },
+    ClearBuffer {
+        dst: Arc<Buffer>,
+        offset: wgt::BufferAddress,
+        size: Option<wgt::BufferAddress>,
+    },
+    ClearTexture {
+        dst: Arc<Texture>,
+        subresource_range: wgt::ImageSubresourceRange,
+    },
+    WriteTimestamp {
+        query_set: Arc<QuerySet>,
+        query_index: u32,
+    },
+    ResolveQuerySet {
+        query_set: Arc<QuerySet>,
+        start_query: u32,
+        query_count: u32,
+        destination: Arc<Buffer>,
+        destination_offset: wgt::BufferAddress,
+    },
+    PushDebugGroup(String),
+    PopDebugGroup,
+    InsertDebugMarker(String),
+    RunComputePass {
+        pass: super::BasePass<super::ArcComputeCommand, Infallible>,
+        timestamp_writes: Option<super::ArcPassTimestampWrites>,
+    },
+    RunRenderPass {
+        pass: super::BasePass<super::ArcRenderCommand, Infallible>,
+        color_attachments: super::ArcRenderPassColorAttachmentArray,
+        depth_stencil_attachment: Option<super::ArcRenderPassDepthStencilAttachment>,
+        timestamp_writes: Option<super::ArcPassTimestampWrites>,
+        occlusion_query_set: Option<Arc<QuerySet>>,
+    },
+    BuildAccelerationStructures {
+        blas: Vec<crate::ray_tracing::ArcBlasBuildEntry>,
+        tlas: Vec<crate::ray_tracing::ArcTlasPackage>,
+    },
+    TransitionResources {
+        buffer_transitions: Vec<wgt::BufferTransition<Arc<Buffer>>>,
+        texture_transitions: Vec<wgt::TextureTransition<Arc<Texture>>>,
+    },
+}